cirq/circuits/expectation_value.py

```diff
@@ -0,0 +1,340 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Expectation value tool for pauli operators given a circuit"""
+
+# imports
+from typing import Dict, Union
+
+import numpy as np
+from cirq.google import XmonSimulator, XmonMeasurementGate
+from cirq.circuits import Circuit
+from cirq.ops import RotXGate, RotYGate, MeasurementGate, Pauli, PauliString
+from cirq.devices import GridQubit
+from cirq.line import LineQubit
+from cirq.ops import NamedQubit
+
+
+# Helper Function to add measurement gates to qubits
+def measurement_gates(circuit: Circuit,
+                      q_dict: Dict[Union[NamedQubit, GridQubit, LineQubit],
```

Comment:
Is it actually important that it be these three specific kinds of qubits? If not, use `ops.QubitId` instead of the union.